apps/examples/stdk_smart_switch/smart_switch.c

```diff
@@ -0,0 +1,304 @@
+	}
+
+	SL_LOGI(TAG, "Sequence number return : %d", sequence_no);
+	st_cap_attr_free(on_evt);
+}
+
+void cap_switch_noti_cb(iot_noti_data_t *noti_data, void *noti_usr_data)
+{
+	SL_LOGI(TAG, "Notification message received");
+
+	if (noti_data->type == IOT_NOTI_TYPE_DEV_DELETED) {
+		SL_LOGI(TAG, "[device deleted]");
+	} else if (noti_data->type == IOT_NOTI_TYPE_RATE_LIMIT) {
+		SL_LOGI(TAG, "[rate limit] Remaining time:%d, sequence number:%d",
+			noti_data->raw.rate_limit.remainingTime, noti_data->raw.rate_limit.sequenceNumber);
+	}
+}
+
+static void smartswitch_task(void *arg)
+{
+	IOT_CAP_HANDLE *handle = (IOT_CAP_HANDLE *)arg;
+
+	int button_event_type;
+	int button_event_count;
+
+	for (;;) {
+		if (get_button_event(&button_event_type, &button_event_count)) {
+			button_event(handle, button_event_type, button_event_count);
+		}
+
+		if (noti_led_mode != LED_ANIMATION_MODE_IDLE) {
+			change_led_state(noti_led_mode);
+		}
+		vTaskDelay(100 / portTICK_PERIOD_MS);
+	}
+}
+
+#ifdef CONFIG_BUILD_KERNEL
+int main(int argc, FAR char *argv[])
+#else
+void smart_switch_main(void)
+#endif
+{
+	/**
+	  SmartThings Device Kit(STDK) aims to make it easier to develop IoT devices by providing
+	  additional st_iot_core layer to the existing chip vendor SW Architecture.
+
+	  That is, you can simply develop a basic application by just calling the APIs provided by st_iot_core layer
+	  like below. st_iot_core currently offers 14 API.
+
+	  //create a iot context
+	  1. st_conn_init();
+
+	  //create a handle to process capability
+	  2. st_cap_handle_init();
+
+	  //register a callback function to process capability command when it comes from the SmartThings Server.
+	  3. st_cap_cmd_set_cb();
+
+	  //needed when it is necessary to keep monitoring the device status
+	  4. user_defined_task()
+
+	  //process on-boarding procedure. There is nothing more to do on the app side than call the API.
+	  5. st_conn_start();
+	 */
+
+	IOT_CAP_HANDLE* switch_handle = NULL;
+	int iot_err;
+
+	// 1. create a iot context
+	ctx = st_conn_init(onboarding_config, onboarding_config_len, device_info, device_info_len);
+	if (ctx != NULL) {
+		iot_err = st_conn_set_noti_cb(ctx, cap_switch_noti_cb, NULL);
+		if (iot_err)
+			SL_LOGE(TAG, "fail to set notification callback function");
+
+	// 2. create a handle to process capability
+	//	implement init_callback function
+		switch_handle = st_cap_handle_init(ctx, "main", "switch", cap_switch_init_cb, NULL);
+
+	// 3. register a callback function to process capability command when it comes from the SmartThings Server
+	//	implement callback function
+		iot_err = st_cap_cmd_set_cb(switch_handle, "off", cap_switch_cmd_off_cb, NULL);
+		if (iot_err)
+			SL_LOGE(TAG, "fail to set cmd_cb for off");
+		iot_err = st_cap_cmd_set_cb(switch_handle, "on", cap_switch_cmd_on_cb, NULL);
+		if (iot_err)
+			SL_LOGE(TAG, "fail to set cmd_cb for on");
+
+	} else {
+		SL_LOGE(TAG, "fail to create the iot_context");
```

Comment:
Isn't it need "return;" on failure? Is it ok to continue next steps when init or callback setting is failed?

Reply:
yes right, fixed